docs/ci.md

```diff
@@ -0,0 +1,206 @@
+as CI. Github triggers are used to run a subset of the Kubernetes E2E tests on
+each pull request. Below describes how to update the set of tests that run and
+how to run these tests locally.
+
+## Updating CI test suite
+
+The tests that are run in CI are from the generated file:
+* [ovn-kubernetes/.github/workflows/test_generated.yml](https://github.com/ovn-org/ovn-kubernetes/blob/master/.github/workflows/test_generated.yml)
+
+This file is generated from the following two files:
+* [ovn-kubernetes/.github/workflow-templates/render.rb](https://github.com/ovn-org/ovn-kubernetes/blob/master/.github/workflow-templates/render.rb)
+    * This file primarily defines the set of file to be skipped.
+* [ovn-kubernetes/.github/workflow-templates/test.yml.erb](https://github.com/ovn-org/ovn-kubernetes/blob/master/.github/workflow-templates/test.yml.erb)
+    * This file is a template, which creates test steps. Each step can be
+reused in multiple tests.
+    * This file also contains the `kubetest` syntax to run each test.
+
+A newer version `ruby` (ruby-2.6.3 for example) is needed to regenerate the
+`test_generated.yml` file. Below are the steps:
+
+```
+$ cd $GOPATH/src/github.com/ovn-org/ovn-kubernetes/.github/workflow-templates/
+$ ruby render.rb
+```
+
+## Running CI Locally
+
+This section describes how to run CI tests on a local deployment. This may be
+useful for expanding the CI test coverage, or testing a private fix before
+creating a pull request.
+
+### Download and install kubetest
+
+There are a variety of ways to run e2e tests, but the preferred method (and the
+method used by OVN-Kubernetes) is using `kubetest`. CI pulls in a pre-built
+`kubetest` binary (See
+[test_generated.yml](https://github.com/ovn-org/ovn-kubernetes/blob/master/.github/workflows/test_generated.yml))
+built on Fedora. This method can be repeated, or the binary can be built from
+source.
+
+#### Download pre-built kubetest binary
+
+To download the pre-built `kubetest` binary which was built on Fedora:
+
+```
+$ GO111MODULE="on"
+$ mkdir -p $GOPATH/bin
+$ curl -fs https://chunk.io/trozet/ba750701d0af4e2b94b249ab9de27b50 -o $GOPATH/bin/kubetest
+$ chmod +x $GOPATH/bin/kubetest
+   
+$ ls -al $GOPATH/bin
+drwxrwxr-x. 2 user1 user1       34 Mar 16 08:30 .
+drwxrwxr-x. 5 user1 user1       39 Feb  7 17:56 ..
+-rwxrwxr-x. 1 user1 user1 59606466 Mar 16 08:30 kubetest
+```
+
+#### Build kubetest from source
+
+To install `kubetest`:
+
+```
+$ mkdir -p $GOPATH/bin
+
+$ cd $GOPATH/src
+$ go get github.com/kubernetes/test-infra; cd github.com/kubernetes/test-infra
+$ GO111MODULE=on go install ./kubetest
+   
+$ ls -al $GOPATH/bin
+drwxrwxr-x. 2 user1 user1       34 Mar 16 08:30 .
+drwxrwxr-x. 5 user1 user1       39 Feb  7 17:56 ..
+-rwxrwxr-x. 1 user1 user1 59606466 Mar 16 08:30 kubetest
+```
+
+### Download and build kubernetes
+
+Kubernetes v1.16.4, which is the version used in CI, does not build with go
```

Comment:
I would prefer we don't put versions in this doc. Otherwise we will have to remember to update them. Maybe we can just say use the K8S_VERSION var to control which version of k8s is used and ensure you build your kubeclient to match.

I think I'm going to pursue  https://github.com/ovn-org/ovn-kubernetes/pull/1112 further and merge it wish skipping the unsupported test case so we can move to a more recent version of k8s.

Reply:
Done.